lib/web/desktop.go

```diff
@@ -218,7 +226,20 @@ const (
 	SNISuffix = ".desktop." + constants.APIDomain
 )
 
-func (h *Handler) desktopTLSConfig(ctx context.Context, ws *websocket.Conn, clusterClient auth.ClientI, sessCtx *SessionContext, desktopName, username, siteName string) (_ *tls.Config, err error) {
+// desktopTLSConfig returns a TLS config for connecting to a Windows Desktop Service.
+// It also performs MFA if required.
+//
+// This function returns the TLS config, and a list of withheld TDP messages if MFA was
+// required and any were received before the MFA message was sent. It is the caller's
+// responsibility to send these withheld messages through to the server after the MFA
+// ceremony is completed.
```

Comment:
I think the long confusing comment here is an indicator that `desktopTLSConfig` has become unwieldy. It started out as a way to build a `tls.Config` for the connection, now it withholds messages, does MFA, etc.

At a minimum we should rename it, but I wonder if we can come up with a better design so that we don't have to plumb these withheld messages through.

Reply:
I made `withheldMessages` a member of `Handler` and restructured things to make the steps more readily apparent from the function names: 6aa0331cda8fa820514dc896c6e75a7f56f45b30